taichi/transforms/lower_ast.cpp

```diff
@@ -13,17 +17,27 @@ std::vector<T *> make_raw_pointer_list(
   return raw_pointers;
 }
 
+}  // namespace
+
 // Lower Expr tree to a bunch of binary/unary(binary/unary) statements
 // Goal: eliminate Expression, Identifiers, and mutable local variables. Make
 // AST SSA.
 class LowerAST : public IRVisitor {
  private:
   Stmt *capturing_loop;
   std::unordered_set<Stmt *> detected_fors_with_break;
+  Block *current_block_;
```

Comment:
Why not just use `current_block`?

Reply:
Good point, here's my thought:

1. Google's C++ style recommends private class members to end with `_`, see https://google.github.io/styleguide/cppguide.html#Variable_Names
2. We already have a global `current_block`. Having a member variable could be very confusing.

That said, I know that this class didn't follow 1 when it was created, and that we might want to deprecate the global `curretn_block` asap, so I've renamed this to `current_block`. Moreover, whenever I need to refer to it, I've prepended `this->` to distinguish it from the global one.